### PR TITLE
Add Description of Routes and Entity properties to the REST-API Documentation#4278

### PR DESCRIPTION
**Pull request by barshimi**
Closes: https://github.com/amplication/amplication/issues/4278

### CODE REVIEW

**Review by morhag90 — APPROVED** [review]

File: plugins/swagger-apibody/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    eventParams: CreateEntityControllerBaseParams
+  ) {
+    const { templateMapping, template, controllerBaseId } = eventParams;
+
+    interpolate(template, templateMapping);
+
+    const classDeclaration = getClassDeclarationById(
+      template,
+      controllerBaseId
+    );
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/src/index.ts:40]: @barshimi I dont think you need to interpolate the template and the params 

You should be able to work directly on the template without interpolation

File: plugins/swagger-apibody/src/util/ast.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
+import * as recast from "recast";
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/src/util/ast.ts:1]: There is already a library called code-gen-utils
https://www.npmjs.com/package/@amplication/code-gen-utils

if these functions are missing, please update the library

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Review by tupe12334 — APPROVED** [review]

File: plugins/swagger-apibody/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Scripts
+
+### `build`
+
+Running `npm run build` will bundle your plugin with Webpack for production.
+
+### `dev`
+
+Running `npm run dev` will watch your plugin's source code and automatically bundle it with every change.
+
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/README.md:24]: This is not needed - you are describing the scripts of the plugins, the idea was to describe scripts that the user may need to use

File: plugins/swagger-apibody/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# @amplication/plugin-<name>
+
+[![NPM Downloads](https://img.shields.io/npm/dt/@amplication/plugin-auth-basic)](https://www.npmjs.com/package/@amplication/plugin-auth-basic)
+
+Add Description of Routes and Entity properties to the REST-API Documentation
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/README.md:5]: maybe you can add a screenshot of the actual results in swagger?

Comment (yuval-hazaz) [reply plugins/swagger-apibody/README.md:5]: Also, you can add an example of the actual decorators the plugin is adding

File: plugins/swagger-apibody/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+## Purpose
+
+the plugin add routes body to each controller
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/README.md:9]:
```suggestion
The plugin adds the details of the type of the body to the `Create` and `Update` methods in each controller 
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: plugins/swagger-apibody/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# @amplication/plugin-<name>
```
Comment (abrl91) [inline plugins/swagger-apibody/README.md:1]: update this line with the plugin name

File: plugins/swagger-apibody/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        "eslint": "^8.21.0",
+        "jest-mock-extended": "^3.0.1",
+        "lodash": "^4.17.21",
+        "prettier": "^2.7.1",
+        "recast": "0.20.5",
+        "rimraf": "^4.0.7",
+        "ts-loader": "^9.4.2",
+        "typescript": "4.6.2",
+        "webpack": "^5.75.0",
+        "webpack-cli": "^5.0.1"
```
Comment (abrl91) [inline plugins/swagger-apibody/package.json:33]: do you need all of these dependencies?

File: plugins/swagger-apibody/src/util/ast.ts [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+const CONSTRUCTOR_NAME = "constructor";
+
+export function transformTemplateLiteralToStringLiteral(
+  templateLiteral: namedTypes.TemplateLiteral
+): namedTypes.StringLiteral {
+  const value = templateLiteral.quasis
+    .map((quasie, i) => {
+      const expression = templateLiteral.expressions[
+        i
```
Comment (yuval-hazaz) [inline plugins/swagger-apibody/src/util/ast.ts:14]: There is already a library called code-gen-utils
https://www.npmjs.com/package/@amplication/code-gen-utils

if these functions are missing, please update the library

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: plugins/swagger-apibody/package.json (hunk 7 above)
Comment (barshimi) [reply plugins/swagger-apibody/package.json:33]: yes

File: plugins/swagger-apibody/src/util/ast.ts (hunk 8 above)
Comment (barshimi) [reply plugins/swagger-apibody/src/util/ast.ts:14]: do you mean visit from recast ? cause it is part of the library not [@amplication/code-gen-utils]

File: plugins/swagger-apibody/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+import {
+  getClassDeclarationById,
+  getClassMethodById
+} from "./util/ast";
+import { EventNames } from "@amplication/code-gen-types";
+import { builders, namedTypes } from "ast-types";
+
+const funcMethodMap = {
+  CREATE_ENTITY_FUNCTION: "CREATE_INPUT",
+  UPDATE_ENTITY_FUNCTION: "UPDATE_INPUT"
```
[the remark this answers is not in the document]
Comment (barshimi) [reply plugins/swagger-apibody/src/index.ts:16]: it's getting an existed values, I cant change them

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**